var/spack/repos/builtin/packages/py-qiskit-aer/package.py

```diff
@@ -0,0 +1,52 @@
+# Copyright 2013-2021 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+
+class PyQiskitAer(PythonPackage, CudaPackage):
+    """Aer is a high performance simulator for quantum circuits that
+    includes noise models"""
+
+    homepage = "https://github.com/Qiskit/qiskit-aer"
+    pypi = "qiskit-aer/qiskit-aer-0.9.1.tar.gz"
+
+    version('0.9.1', sha256='3bf5f615aaae7cc5f816c39a4e9108aabaed0cc894fb6f841e48ffd56574e7eb')
+
+    depends_on('python@3.6:', type=('build', 'run'))
+    depends_on('py-setuptools@40.1.0:', type='build')
+    depends_on('py-numpy@1.16.3:', type=('build', 'run'))
+    depends_on('py-pybind11@2.6:', type='build')
+    depends_on('py-qiskit-terra@0.17.0:', type=('build', 'run'))
+    depends_on('py-scipy@1.0:', type=('build', 'run'))
+    depends_on('py-scikit-build@0.11.0:', type='build')
+    depends_on('py-cmake@:3.16,3.18:', type='build')
+    depends_on('mpi')
+    depends_on('nlohmann-json@3.1.1:')
+    depends_on('spdlog@1.5.0:')
+    depends_on('muparserx@4.0.8:')
+    depends_on('blas')
+    depends_on('cuda@10.1:', when='+cuda')
+
+    variant('mpi', default=True, description='Enable MPI support')
+    variant('gdr', default=True, description='Enable GDR support')
+
+    def setup_build_environment(self, env):
+        env.set('DISABLE_CONAN', 'ON')
+        env.set('DISABLE_DEPENDENCY_INSTALL', '1')
+        env.set("CUDAHOSTCXX", spack_cxx)
+
+    def build_args(self, spec, prefix):
+        args = []
+        args.append('-DDISABLE_CONAN=ON')
+        if '~gdr' in self.spec:
+            args.append('-DAER_DISABLE_GDR=True')
+        if '+mpi' in self.spec:
+            args.append('-DAER_MPI=True')
```

Comment:
I prefer to explicitly disable these in an else clause. CMake likes to change the default values based on whether or not the dependencies are found on the system, leading to non-deterministic builds.